packages/middleware/rate-limiter/src/slot.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Slot provider.
+ */
+export interface SlotProvider {
+  /**
+   * Creates an active slot for the given period.
+   */
+  create(args: { fromSecond: number; durationSecond: number; key: string }): Slot
+}
+
+/**
+ * Sliding window slot.
+ */
+export interface Slot {
+  /**
+   * Slot starting time.
+   */
+  readonly fromSecond: number
```

Comment:
It is not worth it in typescript since there's no opaque types, but this is a place where you could use a phantom type in other languages. If you want an interesting read about it I can recommend you [this blog post](https://blog.hayleigh.dev/phantom-types-in-gleam) :)